Tests/js/views/fields/assets/ez-relation-editview-tests.js

```diff
@@ -0,0 +1,419 @@
+            );
+            Y.Assert.areSame(
+                this.view.get('field').fieldValue.destinationContentId,
+                this.view.get('destinationContentId'),
+                "The destinationContentId of the field value should be the same than the attribute"
+            );
+        },
+
+        "Test validate required": function () {
+            var fieldDefinition = this._getFieldDefinition(true);
+
+            this.view.set('fieldDefinition', fieldDefinition);
+            this.view._set('destinationContentId', null);
+            this.view.validate();
+            this.view.render();
+
+            Y.Assert.isFalse(
+                this.view.isValid(),
+                "An empty relation is NOT valid"
+            );
+        },
+
+        "Should fire the loadFieldRelatedContent event": function () {
+            var loadContentEvent = false,
+                that = this;
+
+            this.view.on('loadFieldRelatedContent', function (e) {
+                Y.Assert.areSame(
+                    that.fieldDefinitionIdentifier,
+                    e.fieldDefinitionIdentifier,
+                    "fieldDefinitionIdentifier is the same than the one in the field"
+                );
+                loadContentEvent = true;
+            });
+            this.view._set('destinationContentId', 45);
+            this.view.set('active', true);
+
+            Y.Assert.isTrue(loadContentEvent, "loadContentEvent should be called when changing active value");
+        },
+
+        "Should NOT fire the loadFieldRelatedContent event if field is empty": function () {
+            var loadContentEvent = false,
+                that = this;
+
+            this.view.on('loadFieldRelatedContent', function (e) {
+                Y.Assert.areSame(
+                    that.fieldDefinitionIdentifier,
+                    e.fieldDefinitionIdentifier,
+                    "fieldDefinitionIdentifier is the same than the one in the field"
+                );
+                loadContentEvent = true;
+            });
+            this.field = {fieldValue: {destinationContentId: null}};
+            this.view = new Y.eZ.RelationEditView({
+                container: '.container',
+                field: this.field,
+                fieldDefinition: this.fieldDefinition,
+                content: this.content,
+                version: this.version,
+                contentType: this.contentType,
+                destinationContent: this.destinationContent,
+                destinationContentId: null,
+            });
+
+            this.view.set('active', true);
+
+            Y.Assert.isFalse(loadContentEvent, "loadContentEvent should NOT be called when changing active value");
+        },
+
+        "Should run the UniversalDiscoveryWidget": function () {
+            var that = this;
+
+            this.view.on('contentDiscover', function (e) {
+                that.resume(function () {
+                    Y.Assert.isObject(e.config);
```

Comment:
missing the text and you should also test that the `config` contains the 2 functions under the `contentDiscoveredHandler` and `cancelDisvoverHandler` properties.